### DISCUSSION TITLE
saveState with p.changeVisualShape, color not change

### DISCUSSION THREAD

**Discussion by chao0716** [bulletphysics/bullet3#3379] [General]
Hello,

I found if I load a urdf object and change its color by p.changeVisualShape(Uid, -1, rgbaColor=[np.random.random(), np.random.random(), np.random.random(), 1.0]) and save the state with p.saveBullet(save_dir+'state.bullet'). 

When I load the state again, it will not restore the color. Is it normal and how to restore the state with color change. 

Thanks.

**Comment by erwincoumans** [comment]
Indeed it is expected, saveState only saves partial state, related to dynamics, not visuals.
